File: test/scripts/09-random-io/incorrect.js
const fs = require('fs');
const path = require('path');
const os = require('os');
const SIZE_MB = 20;
const READS = 5000;
const CHUNK = 4096;
const DURATION_SEC = parseInt(process.env.DURATION_SEC || '300', 10);
const tmp = path.join(os.tmpdir(), 'perf-random-io-' + Date.now() + '.dat');
const buf = Buffer.alloc(SIZE_MB * 1024 * 1024);
buf.fill(0);
fs.writeFileSync(tmp, buf);
let laps = 0;
const start = Date.now();
const maxOff = SIZE_MB * 1024 * 1024 - CHUNK;
const fd = fs.openSync(tmp, 'r');
console.log('INCORRECT: Random seeks');
function run() {
  if (Date.now() - start >= DURATION_SEC * 1000) {
    fs.closeSync(fd);
    fs.unlinkSync(tmp);
    console.log('Stopped:', laps, 'laps');
    return;
  }
  for (let i = 0; i < READS; i++) {
    const pos = Math.floor(Math.random() * maxOff);
    const b = Buffer.alloc(CHUNK);
    fs.readSync(fd, b, 0, CHUNK, pos);
  }
  laps++;
  setImmediate(run);
}
run();
process.on('SIGINT', () => { fs.closeSync(fd); fs.unlinkSync(tmp); process.exit(0); });

File: test/scripts/09-random-io/incorrect.py
#!/usr/bin/env python3
"""
INCORRECT: Random I/O - seeks, slow on HDD/SSD.
Resume: "Prefer Sequential I/O"
"""
import os
import random
import signal
import tempfile
import time
SIZE_MB = 20
READS = 5000
CHUNK = 4096
DURATION_SEC = int(os.environ.get("DURATION_SEC", "300"))
stop = False

def main():
    global stop
    print("INCORRECT: Random seeks")
    signal.signal(signal.SIGINT, lambda s, f: globals().__setitem__('stop', True))
    signal.signal(signal.SIGTERM, lambda s, f: globals().__setitem__('stop', True))
    with tempfile.NamedTemporaryFile(delete=False, suffix='.dat') as f:
        path = f.name
    try:
        with open(path, 'wb') as f:
            f.write(os.urandom(SIZE_MB * 1024 * 1024))
        max_off = (SIZE_MB * 1024 * 1024) - CHUNK
        start = time.perf_counter()
        laps = 0
        while not stop and (time.perf_counter() - start) < DURATION_SEC:
            with open(path, 'rb') as f:
                for _ in range(READS):
                    f.seek(random.randint(0, max_off))
                    f.read(CHUNK)
            laps += 1
        print(f"Stopped: {laps} laps in {time.perf_counter()-start:.1f}s")
    finally:
        os.unlink(path)

if __name__ == "__main__":
    main()
